eZ/Publish/Core/REST/Server/Controller/Content.php

```diff
@@ -316,6 +316,46 @@ public function copyContent($contentId, Request $request)
     }
 
     /**
+     * Deletes a translation from all the Versions of the given Content Object.
+     *
+     * If any non-published Version contains only the Translation to be deleted, that entire Version will be deleted
+     *
+     * @param int $contentId
+     * @param string $languageCode
+     *
+     * @return \eZ\Publish\Core\REST\Server\Values\NoContent
+     *
+     * @throws \Exception
+     */
+    public function deleteContentTranslation($contentId, $languageCode)
+    {
+        $contentService = $this->repository->getContentService();
+
+        $this->repository->beginTransaction();
+        try {
+            $contentInfo = $contentService->loadContentInfo($contentId);
+            // Facade: delete all non-published Versions which contain only single language to be removed
+            foreach ($contentService->loadVersions($contentInfo) as $versionInfo) {
+                if (!$versionInfo->isPublished() && $versionInfo->languageCodes === [$languageCode]) {
+                    $contentService->deleteVersion($versionInfo);
+                }
+            }
```

Comment:
I'm not following the "facade" thing. What does it refer to exactly ?

Reply:
> I'm not following the "facade" thing. What does it refer to exactly ?

Facade in the sense of doing things for REST API Consumer. We have quite common use case: you've created in PlatformUI `eng-GB` version, added Translation, e.g. `fre-FR`, changed Main Translation to `fre-FR`. Now you want to delete `eng-GB` Translation from all Versions. Normally you'd be required to manually delete entire Version no. 1 as it contains only single `eng-GB` Translation. 

This facade does that for you.

I know this does not strictly follow reflecting PHP API in REST API, however we had a lot of complaints from Customers about this feature (we were able to convince them why they need to change Main Translation manually, however we have no arguments when it comes to deleting archived Version containing only one translation - we just want to get rid of them as removing single Translation for such Versions does not make sense).
It also simplifies a lot front-end implementation of PlatformUI feature for removing Translation.

So it was decided that REST API could be implemented in an improved way.
// cc @andrerom

Reply:
Why is this not done by the Public API ?
If I understand correctly, if you have `v1 [eng-GB]`, v2 [eng-GB, fre-FR]`, and the main language set to `fre-FR` on v2, you can't delete `eng-GB` from all versions with the Public API ?

Reply:
> Why is this not done by the Public API ?
> If I understand correctly, if you have v1 [eng-GB], v2 [eng-GB, fre-FR], and the main language set tofre-FRon v2, you can't deleteeng-GB` from all versions with the Public API ?

That's right. I could tell you that I was concerned with responsibility of the API, but the truth is I wasn't aware at that time that this is quite common case. Every time I used API with that feature, my first version of a Content always contained two languages. So I just added some validation to make sure we won't leave db in a bad state (like having in lang mask value 0 or 1).
Short: Lack of imagination on my part.

> I'd consider that we change the Public API to do that.

Actually, we can change behavior of `deleteTranslation` as it was introduced in unreleased `6.13` (previously was named `removeTranslation` and changed due to inconsistent naming).
So if we make a decision it has to be done ASAP.

> In terms of code, I'd abstract that particular operation, so that we only call deleteTranslation(). The cleanest would be a ContentService proxy service that overrides that method and takes care of it.

Both ways are fine, for me quicker solution would be in PAPI though.

POV ping @andrerom

Reply:
> Short: Lack of imagination on my part.

Don't blame yourself. Having the whole thing in mind is like 50% of the job. That's why we have an agile development method :)

It would be fantastic if it could be changed before BC hits the game.

Reply:
At the time where this was dicussed we opted to stay explicit in PHP API, and @bdunogier as you yourself are often in favour for moving REST in direction of being higher level then PHP API, we identified covering more covering more cases in REST, to avoid roundtrips.

That said, I don't have a strong opinion here. Maybe there is no use case of making sure no versions are deleted when you only asked to delete translation, and if so we should not optimize for it.